tests/safety/common.py

```diff
@@ -69,6 +71,28 @@ def _rx(self, msg):
   def _tx(self, msg):
     return self.safety.safety_tx_hook(msg)
 
+  def _generic_limit_safety_check(self, msg_function, min_allowed_value: float, max_allowed_value: float, min_possible_value: float,
+                                  max_possible_value: float, test_delta: float = 1, inactive_value: float = 0, msg_allowed=True):
+    """
+      Enforces that a signal within a message is only allowed to be sent within a specific range, min_allowed_value -> max_allowed_value.
+      Tests the range of min_possible_value -> max_possible_value with a delta of test_delta.
+      Message is also only allowed to be sent when controls_allowed is true, unless the value is equal to inactive_value.
+      Message is never allowed if msg_allowed is false, for example when stock longitudinal is enabled and you are sending acceleration requests.
+    """
+
+    # Ensure that we at least test the allowed_value range
+    self.assertGreater(max_possible_value, max_allowed_value)
+    self.assertLessEqual(min_possible_value, min_allowed_value)
+
+    for controls_allowed in [False, True]:
+      # enforce we don't skip over 0 or inactive
+      for v in np.concatenate((np.arange(min_possible_value, max_possible_value, test_delta), np.array([0, inactive_value]))):
+        v = round(v, 2)  # floats might not hit exact boundary conditions without rounding
+        self.safety.set_controls_allowed(controls_allowed)
+        should_tx = controls_allowed and min_allowed_value <= v <= max_allowed_value
+        should_tx = (should_tx or v == inactive_value) and msg_allowed
+        self.assertEqual(self._tx(msg_function(v)), should_tx, (controls_allowed, should_tx, v))
+
```

Comment:
I would make this into a new PR, trying to make it as generic as possible and replace duplicate code

Reply:
https://github.com/commaai/panda/pull/1575